### PR TITLE
Feature aggregation data

### PR DESCRIPTION
**Pull request by mnadiyaka**

### CODE REVIEW

File: src/main/java/com/sigma/model/entity/AggregationResult.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.time.LocalDateTime;
+
+@Entity
```
Comment (tarasVarvariuk) [inline src/main/java/com/sigma/model/entity/AggregationResult.java]: Why is this db entity?

File: src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java [hunk 2]
```diff
@@ -39,4 +40,20 @@ public List<QuizResults> findResultsWithCustomQuery(QuizResultsSearchDto data) {
 
         return q.getResultList();
     }
+
+    @Override
+    public List<Object[]> findQuizResultAggregatedData(AggregationStatisticsDto data) {
+        String query;
+        if (data.shouldApplyAggregation()) {
+            query = "SELECT q." + data.getGrouping().getParam() + ", " + data.getAggregation() + "(r.score)" +
```
Comment (tarasVarvariuk) [inline src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java]: Refactor into constant & string.format

**Review by tarasVarvariuk — DISMISSED** [review]

File: src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java (hunk 2 above)
Comment (mnadiyaka) [reply src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java]: done

File: src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java [hunk 3]
```diff
@@ -39,4 +40,20 @@ public List<QuizResults> findResultsWithCustomQuery(QuizResultsSearchDto data) {
 
         return q.getResultList();
     }
+
+    @Override
+    public List<Object[]> findQuizResultAggregatedData(AggregationStatisticsDto data) {
+        String query;
+        if (data.shouldApplyAggregation()) {
+            query = String.format("SELECT q.%s, %s(r.score) FROM quiz_results r JOIN quizzes q on r.quiz_id = q.quiz_id GROUP BY q.%s%s",
```
Comment (tarasVarvariuk) [inline src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java]: Where is optional group by team?

Comment (tarasVarvariuk) [inline src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java]: Why don't you make it a constant and use multiline string?

**Review by tarasVarvariuk — APPROVED** [review]

**Review by tarasVarvariuk — APPROVED** [review]

File: src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java [hunk 4]
```diff
@@ -39,4 +40,24 @@ public List<QuizResults> findResultsWithCustomQuery(QuizResultsSearchDto data) {
 
         return q.getResultList();
     }
+
+    @Override
+    public List<Object[]> findQuizResultAggregatedData(AggregationStatisticsDto data) {
+        String query;
+        if (data.shouldApplyAggregation()) {
+            query = """
+                    SELECT q.%s, %s(r.score) %s
```
Comment (tarasVarvariuk) [inline src/main/java/com/sigma/repository/impl/CustomQuizResultsStatisticsRepoImpl.java:49]: Should be constant

**Pull request merged**